src/main/resources/assets/opencomputers/loot/openos/lib/tty.lua

```diff
@@ -270,6 +270,10 @@ function tty.stream:read()
   tty.window.handler = nil
   handler.index = 0
 
+  if handler.initial then
+    cursor:update(handler.initial)
+  end
+
   while true do
```

Comment:
Some of the fancier REPLs (such as IPython) will automatically indent lines following a block (such as within an `if`/`while`, etc...). I wanted to recreate a similar behaviour in OC. I realise this is pretty niche, so no complaints if it's a little messy.

I did experiment with alternatives (such as initialising the cursor with data), but sadly this is the only place where one can actual display the text to the user.

Reply:
the way the tty read is being developed, the right way to do this might be to create a subtype of the cursor object, one that doesn't actually "return" its input until the code is complete. This custom cursor object could also prefix at will. I'll write up a prototype of this you can play with